src/Parallel/Main.hpp

```diff
@@ -348,6 +373,11 @@ Main<Metavariables>::Main(CkArgMsg* msg) noexcept {
     tuples::get<tmpl::type_<ParallelComponentProxy>>(the_parallel_components) =
         ParallelComponentProxy::ckNew(opts);
   });
+  if constexpr (detail::has_initialize_phase_change_decision_data_v<
+                    Metavariables>) {
+    typename Metavariables::initialize_phase_change_decision_data{}(
```

Comment:
I've chosen to make it a static `::apply` function, but I agree this is a piece that might need to change as I put together the design of the option-creatable phase change objects. My plan is for the phase-change objects to be in the cache, and they seem the nicest place to put the initialization of their own tags, so this will probably change to dispatch to the list of those objects.

For the moment, I've added the cache as a parameter in case we want to keep this design but grab the phase change objects out of the cache in the process.